lang/cs/Org.Apache.REEF.Network/Naming/NameCache.cs

```diff
@@ -29,7 +29,7 @@ namespace Org.Apache.REEF.Network.Naming
     /// </summary>
     internal sealed class NameCache
     {
-        private readonly MemoryCache _cache;
+        private readonly GenericInMemoryCache _cache;
```

Comment:
Please describe in JIRA for why to make this change. What is not supported in the original one. Is the new one created by you or something similar to the new version of .Net provided?